pkg/auth_proxy/openstack.go

```diff
@@ -50,35 +125,94 @@ func (oh *OpenStackAuthHandler) ServeHTTP(rw http.ResponseWriter, r *http.Reques
 			username := r.FormValue("username")
 			password := r.FormValue("password")
 
-			ok, err := oh.authenticate(username, password)
+			log.Logger.Debugf("%s is setting as '%s'", SessionUsername, username)
+
+			sess.Values[SessionUsername] = username
+
+			vclient, clientErr := client.NewVisualizationClient(visualizationEndpoint, http.Client{}, openstackToken)
+			if clientErr != nil {
+				log.Logger.Errorf("Error during getting visualization client: %s", clientErr)
+				return
+			}
+			// Get the user by its name if exists
+			userGet, err := vclient.GetUserName(username)
```

Comment:
Let us move all this code below to separate method returning only GafanaUpdateCommand, error